app/models/decidim/access_codes/access_code.rb

```diff
@@ -34,7 +34,11 @@ def self.length
       end
 
       def authorizations
-        Decidim::Authorization.where(name: "access_codes").where("metadata->>'access_code_id' = '?'", id)
+        auths = []
+        Decidim::Authorization.find_each do |auth|
+          auths << auth if auth.metadata["access_code_id"] == id
+        end
+        auths
```

Comment:
We can make this solution in a more "Ruby way" using the [select method](https://www.rubyguides.com/2019/04/ruby-select-method/):

```suggestion
        Decidim::Authorization.select { |auth| auth.metadata["access_code_id"] == id }
```